doc/monitoring-stack-deployment/README.md

```diff
@@ -53,11 +63,142 @@ oc expose service prometheus-operated --hostname prometheus.NAMESPACE_NAME.apps.
 ```
 
 8. Deploy grafana
 
 ```bash
-oc apply -f grafana.yaml
+oc apply -f grafana-v4.yaml
+```
+
+#### OpenShift versions from 4.16+ and 3scale Operator version 2.15+
+
+1. Install Grafana operator v5 from the Operator Hub.
+
+2. Install Prometheus operator v4.10.0 from the Operator Hub.
+
+3. Enable monitoring in the APIManager CR
+```yaml
+apiVersion: apps.3scale.net/v1alpha1
+kind: APIManager
+metadata: ...
+spec:
+  monitoring:
+    enabled: true # <------ here
+  ...
+```
+
+4. Create new service account for Prometheus
+
+```bash
+oc create serviceaccount prometheus-monitoring
+```
+
+5. Create a ClusterRoleBinding to give Prometheus ServiceAccount the RBAC permissions required to scrape metrics
+
+```bash
+cat << EOF | oc create -f -
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: prometheus-monitoring
+subjects:
+  - kind: ServiceAccount
+    name: prometheus-monitoring
+    namespace: <3scale namespace>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-monitoring-view
+EOF
+```
+
+6. Create a token for the prometheus-monitoring ServiceAccount
+
+```bash
+oc create token prometheus-monitoring
+```
+This token will expire, which means the Prometheus will lose access to required resource. You can add the `--duration X<s or m>` to specify how long will the token be valid for. Refer to official OpenShift documentation on managing token lifetime.
+
+7. Update the file `3scale-scrape-configs.yaml` bearer_token field with the token generated from the above.
+
+8. Create additional-scrape-config secret:
+
+```bash
+oc create secret generic additional-scrape-configs --from-file=3scale-scrape-configs.yaml=./3scale-scrape-configs.yaml
+```
+
+9. Deploy prometheus
+
+In `prometheus.yaml` file provided, fill the `spec.externalUrl` field with the external URL. The URL template should be:
+
+```yaml
+spec:
+  ...
+  externalUrl: https://prometheus.NAMESPACE_NAME.apps.CLUSTER_DOMAIN
+```
+
+Then deploy prometheus server:
+
+```bash
+oc apply -f prometheus.yaml
+```
+
+10. Create Prometheus route
+
+```bash
+oc expose service prometheus-operated --hostname prometheus.NAMESPACE_NAME.apps.CLUSTER_DOMAIN
+```
+
+11. Deploy grafana datasource
+
+```bash
+oc apply -f datasource-v5.yaml
+```
+
+12. Deploy grafana
+
+```bash
+oc apply -f grafana-v5.yaml
+```
+
+13. Expose grafana route
+
+```bash
+oc expose service example-grafana-service
 ```
+
+#### Migration from v4 to v5
+
+##### Grafana v4 removal
+
+1. Remove the Grafana custom resource to trigger the deletion of Grafana v4 application along with the service and routes. Double check that the route is removed, if not, remove it manually.
+
+2. Remove the Grafana v4 datasource custom resource
+
+3. Remove the Grafana Operator v4
+
+##### Grafana v5 installation
+
+1. Install Grafana Operator v5
+
+2. Create Grafana v5 datasource (instructions above in `OpenShift versions from 4.16+ and 3scale Operator version 2.15+` section)
```

Comment:
@MStokluska  
1) in every step is written "instructions above in `OpenShift versions from 4.16+ and 3scale Operator version 2.15+` section".  And it looks like it's related to  whole section "Grafana v5 installation" , and if so - seems better write it only once at the beginning of the section.
2) Migration currently can be done only on OSD 4.15, that has both v4 and v5 Grafanas. OSD 4.16 (that we see in our staging environment) does not contains v4. Yes, production env and env of Customers can be differ. But maybe we can say at the beginning of this section, that it expected that Openshift itself already migrated, or somethingh like this. Could you please correct me if I'm wrong.

Reply:
`Another words - Migration can be done ONLY on 4.16 + 3scale-operator 2.15. Is it correct or Not` - no, as mentioned in the overview of the doc: `Grafana v5 is available on OpenShift 4.14+ and supported by 3scale Operator from version 2.15+, it is recommended to use Grafana v5 when possible.`

`Can we say at the beginning of this section, that it expected that Openshift itself already migrated` I don't think this is required since the doc clearly states 2 sections:
1. `OpenShift versions up to and including 4.15`
2. `OpenShift versions from 4.16+ and 3scale Operator version 2.15+`

`Does this means that it will work on both OSD (cloud) and OCP (on prem)` - yes, the change isn't related to the flavor of OpenShift but to the underlying version. With 4.16 the tokens are no more automatically generated for service accounts. Therefore, we need to perform a manual creation of sa + token